src/codegen/compile.ml

```diff
@@ -10224,21 +10633,23 @@ and compile_prim_invocation (env : E.t) ae p es at =
   | NextArrayOffset spacing, [e] ->
     let advance_by =
       match spacing with
-      | ElementSize -> Arr.element_size
-      | One -> 2l (* 1 : Nat *) in
+      | ElementSize
```

Comment:
Maybe we should rename `ElementSize` or discard it completely because (if I understand it correctly), the iterator would now advance by an index of 1 and recompute the word offset (by multiplying the word size) before dereferencing the array element.
(Perhaps, briefly repeating the comment that this index value is untagged (but shifted) could be helpful. Or in other words, it uses the specific tag `0b00` as an optimization.)

Reply:
(Maybe a short comment could be added: The `advance_by` value is not tagged but only shifted as an optimzation because it is directly added to the already tagged previous iterator index.)

Reply:
Done in #4375